lib/spack/llnl/util/tty/__init__.py

```diff
@@ -24,6 +26,7 @@
 _warn_enabled = True
 _error_enabled = True
 indent = "  "
+_osascript = os.path.exists('/usr/bin/osascript')
```

Comment:
I'm new here but is there a reason for not using something like:

```suggestion
_osascript = subprocess.check_output("which osascript; exit 0", stderr=None, shell=True).startswith(os.sep)
```

Reply:
You're right, it might be better to just run `which osascript` and check the return code. Let me try that out.